### ISSUE TITLE
MLSL traing issue

### ISSUE THREAD

**Issue by Rpona** [intel/caffe#98]
I am running multi-node training with MLSL. 
I used "mpirun -n 4 -ppn 1 -machinefile ~/mpd.hosts ./build/tools/caffe train -engine "MKL2017" --solver=models/bvlc_alexnet/solver.prototxt" this command.
I found each node will run caffe in the background, however, the master just printed the information as below and keep the status.  
--
Intel(R) Parallel Studio XE 2017 Update 4 for Linux*
Copyright (C) 2009-2017 Intel Corporation. All rights reserved.
Intel(R) Parallel Studio XE 2017 Update 4 for Linux*
Copyright (C) 2009-2017 Intel Corporation. All rights reserved.
Intel(R) Parallel Studio XE 2017 Update 4 for Linux*
Copyright (C) 2009-2017 Intel Corporation. All rights reserved.
--
The CPU usage and the memory usage of each node are lower than normal training.
I compiled with intel "mpicc" and used intel "impi". 
Are they supported by MLSL?

**Comment by jgong5** [comment]
MLSL package has the built-in intel MPI runtime. Why not use that directly?

**Comment by Rpona** [comment]
@jgong5
Thanks for your reminder.
I used built-in intel MPI runtime. It didn't print anything, but it still keep the status until I used ctrl+C to stop it.
I tried the mlsl_test with the command  "mpirun -n 4 -ppn 1 /opt/intel/mlsl_2017.1.016/test/mlsl_test 1"
and it ran successfully.

**Comment by Rpona** [comment]
Thanks for your reminder.
I used built-in intel MPI runtime. It didn't print anything, but it still keep the status until I used ctrl+C to stop it.
I tried the mlsl_test with the command 
"mpirun -n 4 -ppn 1 /opt/intel/mlsl_2017.1.016/test/mlsl_test 1" and it ran successfully.
But if I tried "mpirun -n 4 -ppn 1 **-machinefile ~/mpd.hosts** /opt/intel/mlsl_2017.1.016/test/mlsl_test 2 1"
The same issue will happened.
It just print "built with MLSL API version: 1.0, used MLSL API version: 1.0" and keep the status

**Comment by Rpona** [comment]
Before using MLSL, it can run well with this command.
By the way, I set "MLSL_NUM_SERVERS=4". 
I found if I unset this environment parameter, they will initial the Neural Network, however,
after initial it will crashed.
I0712 11:39:42.690420 11677 solver.cpp:494] Learning Rate Policy: step
(11677): mlsl_impl.hpp:GetIsolationCommCycles:794: ASSERT 'opIdx < stats.size()' FAILED: invalid operation idx
(11676): mlsl_impl.hpp:GetIsolationCommCycles:794: ASSERT 'opIdx < stats.size()' FAILED: invalid operation idx
(11675): mlsl_impl.hpp:GetIsolationCommCycles:794: ASSERT 'opIdx < stats.size()' FAILED: invalid operation idx
(11678): mlsl_impl.hpp:GetIsolationCommCycles:794: ASSERT 'opIdx < stats.size()' FAILED: invalid operation idx

**Comment by fzou1** [comment]
I believe you compile caffe with PERFORMANCE_MONITORING=1.  pls have a try with "export MLSL_STATS=1", or remove the compilation flag (PERFORMANCE_MONITORING).

**Comment by Rpona** [comment]
Hi @fzou1 
Thanks for your great support, after setting 
MLSL_STATS=1
MLSL_NUM_SERVERS=0
and recompiling without PERFORMANCE_MONITORING=1, multi-node training can run well.

However, there is an issue during training.
When master node is "test"ing, all of them will keep the status.
I use "top" command to find the process is 
4 node training -> master  test , 3 node wait -> 4 node wait forever
The final messages are as below:
 I0719 12:38:19.544409 52403 multi_sync.hpp:108] synchronize_params: bcast
I0719 12:38:19.716327 52403 solver.cpp:570] Iteration 1000, Testing net (#0)
(All of them keep the status and didn't print any result.)
